### ISSUE TITLE
Clementine uses old Windows XP notifications, even on Windows Vista/7/8

### ISSUE THREAD

**Issue by ghost** [clementine-player/Clementine#4507]
![2014-08-27_02-25-27](https://cloud.githubusercontent.com/assets/2244606/4057685/27095edc-2dcc-11e4-9d2b-efb1b745b4d0.png)
Clementine uses the old Balloon style icons even on newer versions of Windows.

**Comment by flying-sheep** [comment]
jup, “OS notifications” should be an available option on windows 8 and trigger [toast notifications](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.notifications.toastnotification.aspx)

**Comment by gfactor** [comment]
Would it be possible to use SnoreNotify to add in support for native Windows notifications?
https://github.com/KDE/snorenotify

**Comment by flying-sheep** [comment]
sounds like a cool project! obv. it would be possible (the snorenotify is LGPL3, this one GPL3), but someone has to do it :laughing:

**Comment by narunlifescience** [comment]
Although I agree on improving windows notification status, I don't think adding an extra dependency is a good idea. that would make clementine more harder to maintain on all platforms... It should nt be hard to implement this without using external packages..

**Comment by flying-sheep** [comment]
[this](https://github.com/clementine-player/Clementine/wiki/Compiling-from-Source#prerequisites) is the current list of dependencies. most of them aren’t cross-platform CMake-built Qt projects like snorenotify.

can you please outline what the problem is in your opinion?

**Comment by hatstand** [comment]
Well it's qt5 for one

**Comment by flying-sheep** [comment]
and clementine still is not? uh…

**Comment by narunlifescience** [comment]
You are right! but adding one more to the list wont be any good. clementine already plays well with notifications in linux & mac os. so the only issue is windows notifications. I will have a look in to the windows notification code. clementine team is now busy with pushing out their new release so they wont be accepting any pull requests for the moment.. once they are done with it, i will make a pull request..

**Comment by flying-sheep** [comment]
yeah, the Qt5 thing is a problem as well. don’t take this wrong, but i’m surprised that clementine is still on Qt4.

**Comment by narunlifescience** [comment]
clementine have a qt5 brach as well... at the moment it is available only for linux..

**Comment by flying-sheep** [comment]
and you’re right: probably we don’t need to support as many APIs as snorenotify. Growl and Snarl are dead, now that everything has a dedicated API. so really native APIs + fallback are sufficient.

still, i don’t like that there is so much duplicated work.

ideally there’s be a minimal crossplatform notification lib that just supports freedesktop, OSX, and windows under a common API, and a way to plug-in more backends. then a selection of plugins for toolkit-specific fallbacks (Qt tray bubbles, QML toasts, GTK stuff, …) and mobile platforms.

**Comment by narunlifescience** [comment]
That should be ideal but unfortunately we don't have one..

**Comment by hatstand** [comment]
Easiest thing is probably someone (e.g. me :-) to just build something on top of the windows toast API.

**Comment by narunlifescience** [comment]
That would be great!
